app/vmselect/netstorage/tmp_blocks_file.go

```diff
@@ -35,14 +40,15 @@ func maxInmemoryTmpBlocksFile() int {
 		return 64 * 1024
 	}
 	if maxLen > 4*1024*1024 {
-		return 4 * 1024 * 1024
+		return *tmpBufSize
```

Comment:
I'd suggest to keep previously logic the same and set tmpBufSize=0 by default.

At the top of this function add check:

```go
func maxInmemoryTmpBlocksFile() int {
  if *tmpBufSize > 0 {
       return *tmpBufSize
  }
   ... 
```

 It will allow to overwrite any default values and do not rely on memory.Allowed() limit.